### PR TITLE
docs: Add a roadmap for TUF implementation

### PR DESCRIPTION
**Pull request by jku**
This roadmap should be updated as we go, and each milestone should
get a design document of their own.

Signed-off-by: Jussi Kukkonen <jkukkonen@vmware.com>

-- 
Also updated some details in the baseline doc.

### CODE REVIEW

File: docs/TUF-ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# Milestones
+
+This is a list of achieved milestones and current view of some planned near-future milestones.
+
+* ✓ Milestone 0: baseline repository
+  - An approximation of current repositories without TUF.
+  - A client and a published repository is provided
+  - No tools for editing or maintenance
+  See [BASELINE](docs/BASELINE.md).
```
Comment (lukpueh) [inline docs/TUF-ROADMAP.md:23]: Nit: the intended newline is not rendered.

File: docs/TUF-ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+This means some milestones may not always move the project directly towards the end goal: Easily reachable and functional milestones are more important than progress towards final goal.
+
+There is no need to have every milestone defined in the beginning: It's fine if we can see a few milestones ahead.
+
+# Milestones
+
+This is a list of achieved milestones and current view of some planned near-future milestones.
+
+* ✓ Milestone 0: baseline repository
+  - An approximation of current repositories without TUF.
```
Comment (lukpueh) [inline docs/TUF-ROADMAP.md:20]: Consistency nit: This is the only line in your itemizations here that you end with a period.

**Review by lukpueh — APPROVED** [review]
This is great and should get us moving. I suppose we can always not only add new milestones but update existing ones, right?

**Review by joshuagl — APPROVED** [review]
Good breakdown of the first steps

**Comment by MVrachev** [comment]
Even though I saw it a couple of days later, I agree it's a good breakdown and will be excited to be part of this.

**Pull request merged**